### PR TITLE
[8.13] Data Quality persistence

### LINKED ISSUE #4807: [Request] Document Persisting Data Quality Results
### Description

What: Unlike before, we're finally persisting data quality results
How: When user runs a DQ check, those results are written to an internal index

### Background & resources

* PRs: https://github.com/elastic/kibana/pull/173185
* Issues/metas: https://github.com/elastic/security-team/issues/7382
* Point of contact: @semd @andrew-goldstein @angorayc @dhru42 
* Test environments: 

### Which documentation set does this change impact?

ESS and serverless

### ESS release

8.13

### Serverless release

tbd

### Feature differences

N/A

### API docs impact

tbd

### Prerequisites, privileges, feature flags

_No response_

### PR DESCRIPTION
**Pull request by benironside**
Fixes #4807 by updating the data quality dashboard doc to inform users about the updates which make the data quality information persist between sessions and across users.

Preview: [Data quality dashboard](https://security-docs_bk_4953.docs-preview.app.elstc.co/guide/en/security/master/data-quality-dash.html)

### CODE REVIEW

File: docs/dashboards/data-quality-dashboard.asciidoc [hunk 1]
```diff
@@ -29,14 +29,16 @@ To use the Data Quality dashboard, you need at least the following {ref}/securit
 [discrete]
 [[data-quality-dash-check-indices]]
 == Check indices
-When you open the dashboard, data does not appear until you select indices to check.
+When the dashboard gets opened for the first time in a project, data does not appear until you (or a teammate) select indices to check.
```
Comment (nastasha-solomon) [inline docs/dashboards/data-quality-dashboard.asciidoc:32]: Minor revisions to keep the voice active. 

```suggestion
When someone opens the dashboard for the first time in a project, data does not appear until you (or another user) select indices to check. 
```

File: docs/dashboards/data-quality-dashboard.asciidoc [hunk 2]
```diff
@@ -29,14 +29,16 @@ To use the Data Quality dashboard, you need at least the following {ref}/securit
...
-When you open the dashboard, data does not appear until you select indices to check.
+When the dashboard gets opened for the first time in a project, data does not appear until you (or a teammate) select indices to check. 
 
 * *Check multiple indices*: To check all indices in the current data view, click *Check all* at the top of the dashboard. A progress indicator will appear.
 
 IMPORTANT: To customize which indices are checked when you click *Check all*, {security-guide}/data-views-in-sec.html[change the current data view].
 
 * *Check a single index*: To check a single index, expand it using the arrow on the left. Checking a single index is faster than checking all indices.
 
+Once checked, an index's data quality information persists indefinitely. You can see when it was last checked, and recheck it at any time.
```
Comment (nastasha-solomon) [inline docs/dashboards/data-quality-dashboard.asciidoc:40]: I'm not entirely clear what "information" means in the first sentence. If you're referring to the results of the data quality check, maybe swap out "information" with "results" or "details" instead.

```suggestion
Once checked, an index's data quality results persist indefinitely. You can see when it was last checked, and recheck it at any time.
```

**Review by nastasha-solomon — APPROVED** [review]
Just two minor suggestions. Everything LGTM otherwise!

File: docs/dashboards/data-quality-dashboard.asciidoc [hunk 3]
```diff
@@ -29,14 +29,18 @@ To use the Data Quality dashboard, you need at least the following {ref}/securit
...
 
 * *Check multiple indices*: To check all indices in the current data view, click *Check all* at the top of the dashboard. A progress indicator will appear.
 
 IMPORTANT: To customize which indices are checked when you click *Check all*, {security-guide}/data-views-in-sec.html[change the current data view].
 
 * *Check a single index*: To check a single index, expand it using the arrow on the left. Checking a single index is faster than checking all indices.
 
+Once checked, an index's data quality results persist indefinitely. You can see when it was last checked, and recheck it at any time.
+
+IMPORTANT: Data quality results are stored in a data stream using the following index pattern: `kibana-data-quality-dashboard-results-<spaceId>`, where `<spaceId>` is the ID of the active {kib} {kibana-ref}/xpack-spaces.html[space]. For example, results from the default space are stored in: `kibana-data-quality-dashboard-results-default`.
```
Comment (semd) [inline docs/dashboards/data-quality-dashboard.asciidoc:42]: The data stream name should start with a dot: `.kibana-data-quality-dashboard-results-default`

```suggestion
IMPORTANT: Data quality results are stored in a data stream using the following index pattern: `.kibana-data-quality-dashboard-results-<spaceId>`, where `<spaceId>` is the ID of the active {kib} {kibana-ref}/xpack-spaces.html[space]. For example, results from the default space are stored in: `.kibana-data-quality-dashboard-results-default`. 
```

File: docs/dashboards/data-quality-dashboard.asciidoc [hunk 4]
```diff
@@ -29,14 +29,18 @@ To use the Data Quality dashboard, you need at least the following {ref}/securit
...
-When you open the dashboard, data does not appear until you select indices to check.
+When you open the dashboard for the first time in a project, data does not appear until you (or another user) selects indices to check. 
 
 * *Check multiple indices*: To check all indices in the current data view, click *Check all* at the top of the dashboard. A progress indicator will appear.
 
 IMPORTANT: To customize which indices are checked when you click *Check all*, {security-guide}/data-views-in-sec.html[change the current data view].
 
 * *Check a single index*: To check a single index, expand it using the arrow on the left. Checking a single index is faster than checking all indices.
 
+Once checked, an index's data quality results persist indefinitely. You can see when it was last checked, and recheck it at any time.
```
Comment (dhru42) [inline docs/dashboards/data-quality-dashboard.asciidoc:40]: `You can see when it was last checked, and recheck it at any time.`

what does this refer to? the data quality check or the ability to check persisted results?

File: docs/dashboards/data-quality-dashboard.asciidoc [hunk 5]
```diff
@@ -29,14 +29,18 @@ To use the Data Quality dashboard, you need at least the following {ref}/securit
 [discrete]
 [[data-quality-dash-check-indices]]
 == Check indices
-When you open the dashboard, data does not appear until you select indices to check.
+When you open the dashboard for the first time in a project, data does not appear until you (or another user) selects indices to check.
```
Comment (dhru42) [inline docs/dashboards/data-quality-dashboard.asciidoc:32]: do we need to wordsmith this?

File: docs/dashboards/images/data-qual-dash-detail.png
Comment (dhru42) [inline docs/dashboards/images/data-qual-dash-detail.png:1]: the older photo showed many incompatibilities highlighting the need to use DQ Dashboard

File: docs/dashboards/images/data-qual-dash.png
Comment (dhru42) [inline docs/dashboards/images/data-qual-dash.png:1]: the older photo highlights the large distribution of incompatibilities driving home how users may distinguish between compatible vs. incompatible on the treemap.

File: docs/dashboards/data-quality-dashboard.asciidoc (hunk 5 above)
Comment (joepeeples) [inline docs/dashboards/data-quality-dashboard.asciidoc:32]: This is almost a nit that I let go, but it feels confusing. If I'm the user opening the dashboard, where is "another user" going to come from and select an index before I do? This edit focuses on what the reader alone is doing:
```suggestion
When you open the dashboard for the first time in a project, data does not appear until you select indices to check. 
```

Or, is the behavior actually based on _any user_ selecting an index _at any point_, in which case we don't need to mention opening the dashboard?
```suggestion
Data does not appear in the dashboard until a user selects indices to check. 
```

File: docs/dashboards/data-quality-dashboard.asciidoc (hunk 4 above)
Comment (benironside) [reply docs/dashboards/data-quality-dashboard.asciidoc]: Hopefully it's more clear now!

File: docs/dashboards/images/data-qual-dash-detail.png
Comment (benironside) [reply docs/dashboards/images/data-qual-dash-detail.png:1]: Good point, updated it to include the old data but the new UI element!

File: docs/dashboards/images/data-qual-dash.png
Comment (benironside) [reply docs/dashboards/images/data-qual-dash.png:1]: Also updated this one to include the old data but the new UI element!

File: docs/dashboards/data-quality-dashboard.asciidoc (hunk 5 above)
Comment (benironside) [reply docs/dashboards/data-quality-dashboard.asciidoc]: this is great, thanks Joe

**Review by joepeeples — APPROVED** [review]

**Review by semd — APPROVED** [review]
LGTM

**Review by jmikell821 — APPROVED** [review]
LGTM!

**Pull request merged**